lib/dataSource.js

```diff
@@ -229,203 +220,296 @@ kettle.dataSource.set = function (that, directModel, model, options) {
     return togo;
 };
 
-/**
- * Errback component that handles errors when interacting with dataSource by
- * completing client request (through firing request's onError event).
- */
-// TODO: Incorrect design which couples kettle's request architecture to dataSource
-// error architecture. This should be broadcast down by a Kettle server, as well as 
-// simply using a namespaced listener rather than a whole invoker
-fluid.defaults("kettle.dataSource.errback", {
-    gradeNames: ["autoInit", "fluid.eventedComponent"],
-    events: {
-        onError: "{dataSource}.events.onError"
-    },
-    listeners: {
-        onError: "{that}.handleError"
-    },
+/**** FILE DATASOURCE SUPPORT ****/
+
+fluid.defaults("kettle.dataSource.file", {
+    gradeNames: ["kettle.dataSource"],
+    readOnlyGrade: "kettle.dataSource.file",
+    invokers: {
+        getImpl: {
+            funcName: "kettle.dataSource.file.handle",
+            args: ["{that}", "{arguments}.0", "{arguments}.1"] // options, directModel
+        }
+    }
+});
+
+fluid.defaults("kettle.dataSource.file.writable", {
+    gradeNames: ["kettle.dataSource.writable"],
     invokers: {
-        handleError: {
-            funcName: "kettle.dataSource.errback.handleError",
-            args: [
-                "{requestProxy}",
-                "{arguments}.0"
-            ]
+        setImpl: {
+            funcName: "kettle.dataSource.file.handle",
+            args: ["{that}", "{arguments}.0", "{arguments}.1", "{arguments}.2"] // options, directModel, model
         }
     }
 });
 
-kettle.dataSource.errback.handleError = function (requestProxy, data) {
-    requestProxy.events.onError.fire(data);
+kettle.dataSource.file.handle = function (that, requestOptions, directModel, model) {
```

Comment:
It would be useful to go through and add JSDoc comments to the undocumented functions.